### DISCUSSION TITLE
Dual Island Membership

### DISCUSSION THREAD

**Discussion by JustOneMoreBlock** [BG-Software-LLC/SuperiorSkyblock2#2129] [Pending Feature Requests]
### Description:
I would like to request a feature that allows server owners to enable dual island membership and ownership permissions in the SkyBlock plugin, configurable via the `config.yml` file. This feature should allow players to own and be members of multiple islands. This should also tie in with max amount of team members on an island.

The proposed feature would include:

- **Dual Membership Option**: Allow server owners to enable players to be members of multiple islands, enhancing collaboration.
- **Shared Ownership Option**: Permit server owners to enable players to co-own islands, giving them full management rights while retaining ownership of their original island.
- **Team Limits**: Ensure these features respect server-defined team limits, allowing server owners to control the maximum number of members and owners per island.
- **Toggleable Settings**: Include settings in the `config.yml` that allow server owners to easily toggle these features on or off based on their preferences.

This enhancement would give server owners the flexibility to tailor the multiplayer experience, fostering greater collaboration and allowing friends to work together across multiple islands. Thank you for considering this request.

### Example `config.yml` Settings:

```yaml
# SkyBlock Plugin Configuration

# Allow players to have membership on multiple islands including owning their island.
# Set to true to enable, false to disable.
allow_multiple_island_membership: true
```

This configuration allows for a flexible and customizable setup, providing server owners with control over how players interact with and manage multiple islands.

Allows to do the following:
- /is create (Create your island.)
- /is invite <player> (Invite a friend to your island and won't deny it if you already own an island or are already a member of another island.)
- /is promote <player> (Change the rank of a player on your island.)

**Comment by OmerBenGera** [comment]
Already possible technically using the `/is perms` system
You'll still be part only of one island only, but may have full access to other islands

**Reply by JustOneMoreBlock** [reply]
> ###
> [ <img alt="" width="32" height="32" src="https://avatars.githubusercontent.com/u/9929651?s=64&amp;v=4"> OmerBenGera ](/OmerBenGera) [ Jul 5, 2024 ](#discussioncomment-9969844)
> Maintainer

The only thing I see would be `All` (1st Setting) or `Invite Member` and I'm not seeing any of these allowing me for what I want to do.

**Reply by OmerBenGera** [reply]
> > [ <img alt="" width="32" height="32" src="https://avatars.githubusercontent.com/u/9929651?s=64&amp;v=4"> OmerBenGera ](/OmerBenGera) [ Jul 5, 2024 ](#discussioncomment-9969844)
> > Maintainer
> 
> The only thing I see would be `All` (1st Setting) or `Invite Member` and I'm not seeing any of these allowing me for what I want to do.

I am planning on creating a module (paid module) that will give you access to "island profiles" - you will be able to change between profiles, then being able to be in multiple islands at the same time.

**Reply by JustOneMoreBlock** [reply]
Do you have an ETA on this paid module?

**Reply by OmerBenGera** [reply]
> Do you have an ETA on this paid module?

Sadly no, hopefully in the next few months

**Reply by JustOneMoreBlock** [reply]
Any additional updates to your paid module?

**Comment by JustOneMoreBlock** [comment]
If no additional updates to a paid module. Perhaps any updates of adding this as a feature?

**Reply by OmerBenGera** [reply]
> If no additional updates to a paid module. Perhaps any updates of adding this as a feature?

Hopwfully will resume the development soon :)

**Reply by klevertouch837-tech** [reply]
Do you think it ever will happen because i would love to have it as a feature
